File: 2020/day1.py
INPUT_FILE = 'inputs/day1_input.txt'


def load_data():
    with open(INPUT_FILE, 'r') as f:
        data = f.readlines()
    data = [int(x) for x in data]
    return data

def find_nums(data):
    data = sorted(data)
    for i in range(len(data)):
        for j in range(i,len(data)):
            if data[i] + data[j] == 2020:
                return data[i] * data[j]

def find_nums_part_2(data):
    data = sorted(data)
    for i in range(len(data)):
        for j in range(i,len(data)):
            for k in range(j,len(data)):
                if data[i] + data[j] + data[k] == 2020:
                    return data[i] * data[j] * data[k]

def main():
    data = load_data()
    mult = find_nums_part_2(data)
    print(mult)


if __name__ == "__main__":
    main()

File: 2020/julia/day1.jl

INPUT_FILE = "inputs/day1_input.txt"

function load_data()
    data = readlines(INPUT_FILE)
    data = [parse(Int32,strip(x)) for x in data]
    return data
end

function find_nums(data)
    for i in 1:length(data)
        for j in i:length(data)
            if data[i] + data[j] == 2020
                return data[i] * data[j]
            end
        end
    end
end

function find_nums_part2(data)
    for i in 1:length(data)
        for j in i:length(data)
            for k in j:length(data)
                if data[i] + data[j] + data[k] == 2020
                    return data[i] * data[j] * data[k]
                end
            end
        end
    end
end


data = load_data()
mult = find_nums_part2(data)
println(mult)